packages/services/schema/src/orchestrators.ts

```diff
@@ -669,11 +673,37 @@ const createFederation: (
         } = await compose(subgraphs);
 
         if (tempResult.type === 'success') {
+          const tagImpl =
+            (resolveName: (name: string) => string) => (documentNode: DocumentNode) => {
+              const tagStrategy = createFederationDirectiveStrategy(resolveName('@tag'));
+
+              const tags = new Set<string>();
+
+              function collectTagsFromDirective(directiveNode: DirectiveNode) {
+                const tag = tagStrategy(directiveNode);
+                if (tag) {
+                  tags.add(tag);
+                }
+              }
+
+              visit(documentNode, {
+                [Kind.DIRECTIVE](directive) {
+                  collectTagsFromDirective(directive);
+                },
+              });
+
+              return Array.from(tags);
+            };
+          const tagSpec = new LinkableSpec('https://specs.apollo.dev/tag', {
+            [FEDERATION_V1]: tagImpl,
+            v0_1: tagImpl,
+            v0_2: tagImpl,
+          });
           const supergraphSDL = parse(tempResult.result.supergraph);
-          const tags = extractTagsFromFederation2SupergraphSDL(supergraphSDL);
```

Comment:
I'm going through to simplify this implementation and I can't figure out why we want to collect all tags used by a supergraph.
I was thinking it may be to optimize the contract lookup but these collected tags dont get used there.

I think I can simplify this implementation by only writing a spec for federation to find included/excluded tags and apply `@inaccessible`.